src/mono/wasm/features.md

```diff
@@ -2,36 +2,125 @@
+
+## Multi-threading
 
-# Multi-threading
 Is enabled by `<WasmEnableThreads>true</WasmEnableThreads>`.
+
 It requires HTTP headers similar to `Cross-Origin-Embedder-Policy:require-corp` and `Cross-Origin-Opener-Policy:same-origin`.
-See also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements
 
-# SIMD - Single instruction, multiple data
+See also [SharedArrayBuffer security requirements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements)
+
+JavaScript interop with managed code via `[JSExport]`/`[JSImport]` on the WebWorker is still in development.
+
+## SIMD - Single instruction, multiple data
+
 Is performance optimization enabled by default. It requires recent version of browser.
-You can disable it by `<WasmEnableSIMD>false</WasmEnableSIMD><WasmBuildNative>true</WasmBuildNative>`.
-[See also](https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md)
 
-# EH - Exception handling
+You can disable it by `<WasmEnableSIMD>false</WasmEnableSIMD>`.
+
+See also WebAssembly proposal [SIMD.md](https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md)
+
+Some older devices or operating systems don't have the necessary CPU instructions to make this optimization bring the expected perf boost.
+
+## EH - Exception handling
 Is performance optimization enabled by default. It requires recent version of browser.
-You can disable it by `<WasmEnableExceptionHandling>false</WasmEnableExceptionHandling><WasmBuildNative>true</WasmBuildNative>`.
-[See also](https://github.com/WebAssembly/exception-handling/blob/master/proposals/exception-handling/Exceptions.md)
 
-# BigInt
+You can disable it by `<WasmEnableExceptionHandling>false</WasmEnableExceptionHandling>`.
+
+See also WebAssembly proposal [Exceptions.md](https://github.com/WebAssembly/exception-handling/blob/master/proposals/exception-handling/Exceptions.md)
+
+## BigInt
 Is required if the application uses Int64 marshaling in JS interop.
-[See also](https://github.com/WebAssembly/JS-BigInt-integration)
+See also WebAssembly proposal [JS-BigInt](https://github.com/WebAssembly/JS-BigInt-integration)
+
+## fetch browser API
 
-# fetch browser API
 Is required if the application uses [HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient)
+
 NodeJS needs to install `node-fetch` and `node-abort-controller` npm packages.
 
-# WebSocket browser API
+## WebSocket browser API
+
 Is required if the application uses [WebSocketClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocket)
+
 NodeJS needs to install `ws` npm package.
 
+## Content security policy
+
+dotnet runtime for wasm is CSP compliant starting from .Net 8, except legacy JS interop methods.
+
+In order to enable it, please set HTTP headers similar to `Content-Security-Policy: default-src 'self' 'wasm-unsafe-eval'`
```

Comment:
We could also specify that wasm-unsafe-eval is not needed if you disable the jiterpreter

Reply:
I think until could could ES6 `import` wasm module, the `wasm-unsafe-eval` is always necessary. Because we just compile some bytes.

See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md#the-wasm-unsafe-eval-source-directive

I was not sure, so I tested it even with `WebAssembly.instantiateStreaming` and still I get `CompileError: WebAssembly.instantiateStreaming(): Refused to compile or instantiate WebAssembly module because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'"`